set.mm

```diff
@@ -355904,6 +356277,35 @@ line segment AB (the hypotenuse), and ` O ` is the signed right angle
       KZEDBMKZJKZOUJUIMKZUJUKMKZOCBMKZUHUJJKZOABCDEFGHIPAUJUIUKADBHFQZADCHGQAEU
       JIUPRSAULUNUMUOADBCHFGTAUOLUJJKZUKMKUMALEUJAUAIUPUBAUQUJUKMAUJUPUCUDUEUFU
       G $.
+
+    $( Equivalence between two ways of expressing ` A ` as an affine
+       combination of ` B ` and ` C ` .  (Contributed by AV, 22-Jan-2023.) $)
+    affineequiv3 $p |- ( ph -> ( A = ( ( ( 1 - D ) x. B ) + ( D x. C ) )
+                                 <-> ( A - B ) = ( D x. ( C - B ) ) ) ) $=
```

Comment:
I remember I did some things like that in my mathbox. Have you had a look to see what's reusable?

Reply:
I actually had a look at ~bj-lineq, and there are good chances that it could be used within the proof of ~itsclc0qlem1. But ~bj-lineq depends on ~bj-rdiv which depends on bj-ldiv which is in a block together with ~bj-mdiv, so it would be necessary to move these theorems, too. Then it would be adequate to move all theorems of this section ""Complex numbers (supplements)" to main, isn't it? Where would you propose to place these theorems?

~bj-lineq, however, is currently only used in the proof of ~bj-lineqi, which itself is not used. Therefore, the only usage of it would be in the proof of ~itsclc0qlem1.

Then I had a look at ~bj-bary1, which is based on ~bj-subcom and ~bj-rdiv at the end. I wonder if it can be used to shorten the proof of ~affinecomb1 (instead of using ~affineequiv4 and ~affineequivne which are based on ~affineequiv3). I'll have a look...

Reply:
I haven't looked precisely, but I would be in favor to move that block, next to the similar theorems in Main. I think they have a natural enough statement and fall into a pattern. Maybe bj-lineqi can stay in the mathbox indeed.

What would be very nice is if you could prove the version ~bary2 in two dimensions. The proof would be much longer. I do not know if at that point it will be shorter to use Carmer's formulas or not. Having ~bary2 would suffice already to talk about triangles in a satisfactory way. (My work on barycentric coordinates actually started with a discussion about defining the interior of a triangle.)

But I also have a more general question: why not do all of this (including all the small lemmas to help computations in CC) for general rings (commutative rings, fields when needed) and modules/vector spaces instead ? This will require more groundwork but would certainly be nicer

Reply:
> I haven't looked precisely, but I would be in favor to move that block, next to the similar theorems in Main. I think they have a natural enough statement and fall into a pattern. Maybe bj-lineqi can stay in the mathbox indeed.

I'll have alook at this next...
> 
> What would be very nice is if you could prove the version ~bary2 in two dimensions. The proof would be much longer. I do not know if at that point it will be shorter to use Carmer's formulas or not. Having ~bary2 would suffice already to talk about triangles in a satisfactory way. (My work on barycentric coordinates actually started with a discussion about defining the interior of a triangle.)

I think this goes beyond the scope of this PR

> But I also have a more general question: why not do all of this (including all the small lemmas to help computations in CC) for general rings (commutative rings, fields when needed) and modules/vector spaces instead ? This will require more groundwork but would certainly be nicer

This is also a general issue which cannot/should not be solved in this PR.

Reply:
Indeed.  I should open an issue. Non-blocking and you can resolve.

Reply:
Required theorems moved to main and proof of ~itsclc0lem1 shortened by ~(bj-)lineq.